File: integrations/csharp/Oicana/Inputs/BlobInput.cs
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Nodes;

namespace Oicana.Inputs;

/// <summary>
/// A blob input value for template compilation.
/// Use with a dictionary where the key is the input name.
/// </summary>
public class BlobInput
{
    /// <summary>
    /// Construct a new blob input from bytes
    /// </summary>
    [SetsRequiredMembers]
    public BlobInput(byte[] data) : this(data, null) { }

    /// <summary>
    /// Construct a new blob input from bytes with metadata
    /// </summary>
    [SetsRequiredMembers]
    public BlobInput(byte[] data, BlobMeta? meta)
    {
        Data = data;
        Meta = meta?.Build();
    }

    /// <summary>
    /// Binary data
    /// </summary>
    public required byte[] Data { get; init; }

    /// <summary>
    /// Metadata for the blob (e.g., image format)
    /// </summary>
    public JsonNode? Meta { get; init; }
}

File: integrations/java/oicana/src/main/java/com/oicana/BlobInput.java
package com.oicana;

import java.util.Map;

/**
 * A blob input for a template, consisting of binary data and optional metadata.
 *
 * @param data     the binary content of the blob
 * @param metadata optional metadata as key-value pairs (will be serialized to JSON)
 */
public record BlobInput(byte[] data, Map<String, Object> metadata) {
    /**
     * Create a blob input with data and no metadata.
     *
     * @param data the binary content of the blob
     */
    public BlobInput(byte[] data) {
        this(data, null);
    }
}
